test/Backends/LLVM.hs

```diff
@@ -0,0 +1,127 @@
+  (fn ∷ Word32 → IO Word32) ← jit (Config None) mallocFreeModule "test"
+  res ← fn 7
+  43 T.@=? res
+
+mallocFreeModule ∷ LLVM.AST.Module
+mallocFreeModule =
+  Module
+    "mallocFreeModule"
+    "mallocFreeModule"
+    Nothing
+    Nothing
+    [ GlobalDefinition $
+        functionDefaults
+          { G.returnType = voidStarTy,
+            G.name = Name "malloc",
+            G.parameters = ([Parameter (IntegerType {typeBits = 64}) (Name "size") []], False),
+            G.callingConvention = CC.Fast,
+            G.basicBlocks = [],
+            G.linkage = L.External
+          },
+      GlobalDefinition $
+        functionDefaults
+          { G.returnType = voidTy,
+            G.name = Name "free",
+            G.parameters = ([Parameter voidStarTy (Name "") []], False),
+            G.callingConvention = CC.Fast,
+            G.basicBlocks = [],
+            G.linkage = L.External
+          },
+      GlobalDefinition $
+        functionDefaults
+          { G.returnType = i32,
+            G.name = Name "test",
+            G.parameters = ([Parameter i32 (Name "bar") []], False),
+            G.basicBlocks =
+              [ BasicBlock
+                  (UnName 0)
+                  [ UnName 1 := Call
+                      { tailCallKind = Nothing,
+                        I.function =
+                          Right
+                            ( ConstantOperand
+                                ( C.GlobalReference
+                                    (ptr $ FunctionType {resultType = voidStarTy, argumentTypes = [IntegerType {typeBits = 64}], isVarArg = False})
+                                    (Name "malloc")
+                                )
+                            ),
+                        callingConvention = CC.C,
+                        returnAttributes = [],
+                        arguments = [(ConstantOperand (C.Int {C.integerBits = 64, C.integerValue = 10}), [])],
+                        functionAttributes = [],
+                        metadata = []
+                      },
+                    Do $ Call
+                      { tailCallKind = Nothing,
+                        I.function =
+                          Right
+                            ( ConstantOperand
+                                ( C.GlobalReference
+                                    (ptr $ FunctionType {resultType = voidTy, argumentTypes = [voidStarTy], isVarArg = False})
+                                    (Name "free")
+                                )
+                            ),
+                        callingConvention = CC.C,
+                        returnAttributes = [],
+                        arguments = [(LocalReference voidStarTy (UnName 1), [])],
+                        functionAttributes = [],
+                        metadata = []
+                      }
+                  ]
+                  ( Do $ Ret (Just (ConstantOperand (C.Int 32 43))) []
+                  )
```

Comment:
Would be nice to load the result somewhere, and return it in the final Ret call. Would be nice to show off since malloc gives us backa  Void Pointer.